### PR TITLE
Update Jest tests for schedule veteran workflow

### PR DESCRIPTION
**Pull request by sahalliburton**
### Description
Stacked PR Part 2 for #15212 

Updates and adds Jest tests for the Schedule Veteran Workflow

### CODE REVIEW

File: client/app/components/DataDropdowns/AppealHearingLocations.jsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ AppealHearingLocationsDropdown.propTypes = {
       PropTypes.shape({
         label: PropTypes.string,
         value: PropTypes.shape({
-          facilityId: PropTypes.number,
+          facilityId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (sahalliburton) [inline client/app/components/DataDropdowns/AppealHearingLocations.jsx:174]: Fixes warning about prop-types

File: client/test/app/hearings/components/ScheduleVeteran.test.js [hunk 2]
```diff
@@ -1,41 +1,78 @@
 import React from 'react';
-import { shallow, mount } from 'enzyme';
+import { mount } from 'enzyme';
```
Comment (sahalliburton) [inline client/test/app/hearings/components/ScheduleVeteran.test.js:2]: Most of the changes in this file are due to needing to mount the component which required changing the queueStore to have the state we need

File: client/test/data/appeals.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
+import { hearingDateOptions } from './hearings';
+
 export const defaultAssignHearing = {
   regionalOffice: null,
   hearingLocation: null,
+  hearingDay: null,
```
Comment (sahalliburton) [inline client/test/data/appeals.js:6]: Change the default to show null instead of undefined for hearingDay

File: client/test/data/appeals.js [hunk 4]
```diff
@@ -136,18 +139,7 @@ export const scheduleHearingDetails = {
...
-    timezone: 'America/New_York',
-    scheduledFor: '2020-08-17',
-    requestType: 'V',
-    room: '1',
-    roomLabel: '1 (1W200A)',
-    filledSlots: 2,
-    totalSlots: 12,
-    hearingDate: '2020-08-17'
-  },
+  hearingDay: hearingDateOptions[1].value,
```
Comment (sahalliburton) [inline client/test/data/appeals.js:142]: Replace the dummy data with the shared hearingDateOptions value

File: client/test/data/appeals.js [hunk 5]
```diff
@@ -171,3 +163,11 @@ export const scheduleHearingDetails = {
   }
 }
 ;
+
+export const legacyAppeal = {
```
Comment (sahalliburton) [inline client/test/data/appeals.js:167]: Creates dummy data for legacy appeal

File: client/test/data/hearings.js [hunk 6]
```diff
@@ -446,5 +449,21 @@ export const hearingDateOptions = [
       hearingDate: '2020-10-04'
     }
   }
-]
-;
+];
+
+export const scheduledHearing = {
```
Comment (sahalliburton) [inline client/test/data/hearings.js:454]: Data formatted like the API response

File: client/test/data/tasks.js [hunk 7]
```diff
@@ -16,4 +16,89 @@ export const task = {
   user: 'a'
 };
 
+export const scheduledHearingTask = {
```
Comment (sahalliburton) [inline client/test/data/tasks.js:19]: Mocked hearing task data for tests retrieved from the JavaScript console

**Review by ferristseng — APPROVED** [review]
LGTM, but when I run these tests locally, I'm getting failures because of some snapshot differences. They appear to be the ones failing on CircleCI too, and they aren't resolved after I update the snapshots

**Pull request merged**